### DISCUSSION TITLE
gauge install java with proxy

### DISCUSSION THREAD

**Discussion by kga676** [getgauge/gauge#2114] [General]
Is there any way to execute "gauge install java" on CantOS befind the http proxy?
"export HTTP_PROXY=http://server-ip:port" does not work

**Comment by zabil** [comment]
Setting the `HTTP_PROXY` is the right way to use it. You might want to check of the proxy server has some kind of authentication. Alternatively you can install the plugin [offline](https://docs.gauge.org/plugin.html?os=macos&language=javascript&ide=vscode#install-plugins-offline) by downloading the plugin zip files for your platform.

**Comment by kga676** [comment]
I had already set HTTP_PROXY
and HTTPS_PROXY in maven and in linux
same proxies wors for slack  as
 proxies = {
        'http': 'http://domen.ru:port',
        'https': 'http://domen.ru:ru:port'

without any username\pass
and i still get error

[]# /usr/local/bin/gauge install java
Failed to install plugin 'java'.
Reason: Invalid plugin name or there's a network issue while fetching plugin details.

**Reply by zabil** [reply]
Try running the command `/usr/local/bin/gauge install java -l debug` for more info.

**Reply by kga676** [reply]
Gathering metadata for java
Downloading https://downloads.gauge.org/plugin/java?l=&p=&o=linux&a=amd64
Failed to download java-install.json file: Get "https://downloads.gauge.org/plugin/java?l=&p=&o=linux&a=amd64": Forbidden
Failed to install plugin 'java'.
Reason: Invalid plugin name or there's a network issue while fetching plugin details.

**Reply by sriv** [reply]
Can you check if you are able to `curl https://downloads.gauge.org/plugin/java?l=&p=&o=linux&a=amd64` after setting `HTTP_PROXY` and `HTTPS_PROXY`?

**Reply by kga676** [reply]
Got that

curl https://downloads.gauge.org/plugin/java?l=&p=&o=linux&a=amd64
[1] 18073
[2] 18074
[3] 18075
[]# {
    "versions": [
        {
            "version": "0.7.15",
            "gaugeVersionSupport": {
                "minimum": "1.0.7",
                "maximum": ""
            },
            "install": {
                "windows": [],
                "darwin": [],
                "linux": []
            },
            "DownloadUrls": {
                "x86": {
                    "windows": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-windows.x86.zip",
                    "darwin": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-darwin.x86.zip",
                    "linux": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-linux.x86.zip"
                },
                "x64": {
                    "windows": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-windows.x86_64.zip",
                    "darwin": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-darwin.x86_64.zip",
                    "linux": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-linux.x86_64.zip"
                },
                "arm64": {
                    "windows": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-windows.x86_64.zip",
                    "darwin": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-darwin.x86_64.zip",
                    "linux": "https://github.com/getgauge/gauge-java/releases/download/v0.7.15/gauge-java-0.7.15-linux.x86_64.zip"
                }
            }
        },
        {
            "version": "0.7.13",
            "gaugeVersionSupport": {
                "minimum": "1.0.7",
                "maximum": ""
            },
            "install": {
                "windows": [],
                "darwin": [],
                "linux": []
            },
            "DownloadUrls": {
                "x86": {
                    "windows": "https://github.com/getgauge/gauge-java/releases/download/v0.7.13/gauge-java-0.7.13-windows.x86.zip",
                    "darwin": "https://github.com/getgauge/gauge-java/releases/download/v0.7.13/gauge-java-0.7.13-darwin.x86.zip",
                    "linux": "https://github.com/getgauge/gauge-java/releases/download/v0.7.13/gauge-java-0.7.13-linux.x86.zip"
                },
                "x64": {
                    "windows": "https://github.com/getgauge/gauge-java/releases/download/v0.7.13/gauge-java-0.7.13-windows.x86_64.zip",
                    "darwin": "https://github.com/getgauge/gauge-java/releases/download/v0.7.13/gauge-java-0.7.13-darwin.x86_64.zip",
                    "linux": "https://github.com/getgauge/gauge-java/releases/download/v0.7.13/gauge-java-0.7.13-linux.x86_64.zip"
                }
            }
        },
        {
            "version": "0.7.12",
            "gaugeVersionSupport": {
                "minimum": "1.0.7",
                "maximum": ""
            },
            "DownloadUrls": {
                "x86": {
                    "windows": "https://github.com/getgauge/gauge-java/releases/download/v0.7.12/gauge-java-0.7.12-windows.x86.zip",
                    "darwin": "https://github.com/getgauge/gauge-java/releases/download/v0.7.12/gauge-java-0.7.12-darwin.x86.zip",
                    "linux": "https://github.com/getgauge/gauge-java/releases/download/v0.7.12/gauge-java-0.7.12-linux.x86.zip"
                },
                "x64": {
                    "windows": "https://github.com/getgauge/gauge-java/releases/download/v0.7.12/gauge-java-0.7.12-windows.x86_64.zip",
                    "darwin": "https://github.com/getgauge/gauge-java/releases/download/v0.7.12/gauge-java-0.7.12-darwin.x86_64.zip",
                    "linux": "https://github.com/getgauge/gauge-java/releases/download/v0.7.12/gauge-java-0.7.12-linux.x86_64.zip"
                }
            },
            "install": {
                "windows": [],
                "darwin": [],
                "linux": []
            }
        },
        {

**Reply by kga676** [reply]
Our devops fixed thar by useting proxies and setting them again.

**Reply by sriv** [reply]
Excellent. Thanks for posting back!

**Reply by kga676** [reply]
Now I got other error
I/m trying to 
mvn gauge:execute -DspecsDir=specs/example.spec

gauge installed in /usr/local/bin

error:
[ERROR] Failed to execute goal com.thoughtworks.gauge.maven:gauge-maven-plugin:1.4.3:execute (default-cli) on project autotest-esb: Gauge Specs execution failed. java.io.IOException: Cannot run program "gauge": error=2, No such file or directory -> [Help 1]

**Reply by kga676** [reply]
P.S. already tried https://github.com/getgauge-contrib/gauge-maven-plugin/issues/18
error is thiss apears

**Reply by kga676** [reply]
So, I finnaly found out the problem
I'm working in linux terminal via Putty as user gkopychev
and when I switch to "sudo su" I need to set my proxied AGAIN
and set "export PATH=$PATH:/usr/local/bin" AGAIN
I did it and it started to work: now I got 

Error occurred while waiting for runner process to finish.
Error : exit status 1
Error ----------------------------------

[java]
Failed to start /usr/lib/jvm/java. fork/exec /usr/lib/jvm/java: permission denied

[Gauge]
Failed to start gauge API: Error occurred while waiting for runner process to finish.
Error : exit status 1


but I this this error is not connected wth gauge anymore

**Comment by kga676** [comment]
Damn, when I tried to sudo mvn execute:gauge  I got again


org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.thoughtworks.gauge.maven:gauge-maven-plugin:1.4.3:execute (default-cli) on project autotest-esb: Gauge Specs execution failed. java.io.IOException: Cannot run program "gauge": error=2, No such file or directory


PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin

**Reply by sriv** [reply]
At this point it seems like permissions/env issue. Happy to be a sounding board here, so please feel free to share your investigations and let me know if there's anything you'd like me to help with.

Another thought - have you considered putting the tests in a container, so that you don't need to worry about these every time? Haven't tried this, but if you can get the tests working without proxy (i.e. in your local machine) then you should be able to pass the HTTP_PROXY env to the docker daemon and let it do the proxy routing. This way you can avoid gauge/plugin/maven installation within the proxy and possibly avoid switching users.

**Reply by kga676** [reply]
Thanks alot for a reply, when I'll figure ot out what happens I'll leave my investigation here

**Comment by kga676** [comment]
Solution:
1, Install mvn 3.8.2
2.
sudo su
export HTTPS_PROXY=your_proxy:port
export HTTP_PROXY=your_proxy:port
--gauge to path
export PATH=/usr/local/bin:${PATH}
--mvn to path
export PATH=${M2_HOME}:${PATH}
cd /home/user/at_repo
mvn gauge:execute -DspecsDir=specs


And I got my tests finally!

**Reply by kga676** [reply]
And don't forget to set maven proxies too ( in settings.xml for root too)
